### ISSUE TITLE
Is it a bug to return null when obj instanceof User in SecurityUtil.getUsername()

### ISSUE THREAD

**Issue by donghbcn** [IHTSDO/ihtsdo-spring-sso#4]
Is it a bug to return null when obj instanceof User in SecurityUtil.getUsername()

**Comment by kaicode** [comment]
Hi @donghbcn,

The current logic in the [SecurityUtil.getUsername()](https://github.com/IHTSDO/ihtsdo-spring-sso/blob/3.1.0/ihtsdo-spring-sso/src/main/java/org/ihtsdo/sso/integration/SecurityUtil.java#L10) method (for ihtsdo-spring-sso version 3.1.0) is as follows:
- If the Spring Security Context Authentication object is null, return null
- Else if the Authentication principal object is of type User, then return the username property of the User object
- Else (if principal object is not of type User) then we assume that the principal object is actually the username and we return this object

In your scenario, if the principal object is of type User, then the library returns the username property of the User. If the username property is null then null will be returned. The username property is set by the ([RequestHeaderAuthenticationDecorator](https://github.com/IHTSDO/ihtsdo-spring-sso/blob/3.1.0/ihtsdo-spring-sso/src/main/java/org/ihtsdo/sso/integration/RequestHeaderAuthenticationDecorator.java#L39) using the `X-AUTH-username` HTTP header. I would recommend debugging that decorator class to check that the username header is getting passed in correctly from nginx, otherwise the username will be returned as null.

I hope this helps!

**Comment by donghbcn** [comment]
Thanks for your reply.
I understand the logic. But the 'return ' seems got missing in code as follows:
```Java
        if (obj instanceof User) {
		((User)obj).getUsername();
	} else {
		return (String) obj ;
	}
```

**Comment by kaicode** [comment]
Wow, I see your point!
Will fix.

**Comment by kaicode** [comment]
Fixed and released in 3.1.1.
